modules/caddyhttp/replacer.go

```diff
@@ -208,12 +217,79 @@ func addHTTPVarsToReplacer(repl *caddy.Replacer, req *http.Request, w http.Respo
 	repl.Map(httpVars)
 }
 
+func getReqTLSReplacement(req *http.Request, key *string) (string, bool) {
```

Comment:
Why is `key` a pointer? I believe empty string is also invalid, so there is probably no need for a nil key, right?

Reply:
I'm was trying to reduce memory allocations.  Let me know if you would rather we just pass a string.

Reply:
Let's just pass a string for now; an optimization like this should be profiled first.

Reply:
Fixed.